R/util.R

```diff
@@ -130,27 +130,62 @@ util.is_vector_of_numbers <- function(x){
     # (regardless of whether x is numeric)
     # Character vectors sometimes need to be changed to numerics,
     # e.g., when all columns set to character types by default.
-
+  
+    # determine if x is logical first, because as.numeric will coerce logicals
+    # to boolean 0,1 rather than to NA, meaning util.is_vector_of_numbers would return TRUE
+    if(is.logical(x)) return(FALSE)
+    
     # find numeric values
-    numeric_values <- grepl("^-*[[:digit:]]*\\.*[[:digit:]]+$",x)
+    x_as_numeric <- suppressWarnings(as.numeric(x))
 
-    # find the blank values
-    blank_values <- util.is_blank(x)
+    # anything that gets coerced to NA by as.numeric is not really a number.
+    # but elements that were blank originally are ok.
+    blanks <- is.na(x_as_numeric)
+    coerced_blanks <- blanks & !util.is_blank(x)
 
-    # Return TRUE if all values are either numeric or blank
-    # Unless all values are blank, then false
-    # Return FALSE otherwise
-    if( all(blank_values) ){
+    # Return FALSE if there are any coerced blanks, because that means x 
+    # contained elements that could not be converted to numeric.
+    # Otherwise return TRUE.
+    if(any(coerced_blanks)){
         return(FALSE)
     }
-    else if(all(numeric_values | blank_values)){
+    else{
         return(TRUE)
     }
-    else{
-        return(FALSE)
+}
+
+
+util.is_vector_of_numbers_test <- function(){
+    x_char <- c("1", "a", "b", "1.1", "1e05")
+    x_number <- c("1", "1.1", "2", "-1.1")
```

Comment:
nit: I can think of some other number-like strings we might want to check: `".1"`, `"0.1"`, `"00.1"`,  `"1.00"`, `"+1.1"`, `"01"`.